server/src/main/java/io/littlehorse/App.java

```diff
@@ -15,17 +15,22 @@ public static void doIdempotentSetup(LHConfig config)
         throws InterruptedException, ExecutionException {
         log.info("Creating topics!!");
 
+        boolean createdATopic = false;
         for (NewTopic topic : config.getAllTopics()) {
-            config.createKafkaTopic(topic);
+            createdATopic = config.createKafkaTopic(topic) || createdATopic;
```

Comment:
What's the motivation behind this? how this works? we are creating only one topic?

Reply:
You got the boolean logic backwards. It tries to create all the topics no matter what. If one of them is created, then we wait for 5 seconds to allow the creation to propagate. We already had a sleep for 5 seconds, but now it doesn't sleep if all of the topics already existed.

cc @sauljabin